### DISCUSSION TITLE
What is the right way to get an API reference?

### DISCUSSION THREAD

**Discussion by muxi** [Keyfactor/ejbca-ce#349] [Q&A]
I am using the docker hub image `ejbca-ce` following the `Quick start - ephemeral test instance` instruction. The instance runs fine but the API reference (`/ejbca/doc/rest/`) and Swagger UI (`/ejbca/swagger-ui`) both seem to be missing and returning 404. That's after enabling REST services in System Configuration. Is there something else that can be done or is there an online version of the API? I simply want to see what features are available through the REST API.

**Comment by kumareshsomi** [comment]
Sven pointed out that is an online version accessible [here](https://download.primekey.com/rest/#api-_)
Not sure if that is the latest version. 
There is also reference available in the api and doc dirs of [ejbca-java-client-sdk](https://github.com/Keyfactor/ejbca-java-client-sdk/blob/main/api/openapi.yaml)

**Reply by kumareshsomi** [reply]
And there is the official swagger json: https://github.com/Keyfactor/ejbca-ce/blob/main/modules/ejbca-rest-api/resources/swagger.json

**Comment by primetomas** [comment]
If you want to see java code for calling the REST API there is also the Easy Rest Client. https://github.com/Keyfactor/ejbca-easy-rest-client

**Comment by alex-cbn** [comment]
I have a version of EJBCA from November, and the /ca/ resource under rest-api is nowhere to be found. These are all the protocols available:
![image](https://github.com/Keyfactor/ejbca-ce/assets/17546807/44bc8857-f032-4855-89d9-781cc4f72a3a)
And whenever I run a basic example such as `status` from the documentation, I am greeted by `{"error_code":404,"error_message":"RESTEASY003210: Could not find resource for full path: https://localhost:8443/ejbca/ejbca-rest-api/v1/ca/status"}`.
Does anyone know why this happens?

**Reply by primetomas** [reply]
This is another question from the original one. Please create a new discussion for this question.

Best regards,
Tomas

**Reply by alex-cbn** [reply]
right away. thanks
